src/rdb.c

```diff
@@ -3645,6 +3645,21 @@ void bgsaveCommand(client *c) {
     if (c->argc > 1) {
         if (c->argc == 2 && !strcasecmp(c->argv[1]->ptr, "schedule")) {
             schedule = 1;
+        } else if (c->argc == 2 && !strcasecmp(c->argv[1]->ptr, "cancel")) {
+            /* Terminates an in progress BGSAVE */
+            if (server.child_type == CHILD_TYPE_RDB) {
+                /* There is an ongoing bgsave */
+                serverLog(LL_NOTICE, "Background saving will be aborted due to user request");
+                killRDBChild();
+                addReplyStatus(c, "Background saving cancelled");
+            } else if (server.rdb_bgsave_scheduled == 1) {
+                serverLog(LL_NOTICE, "Scheduled background saving will be cancelled due to user request");
```

Comment:
i see in bgsave schedule, we did not print the server log, it only return a reply, do we need to print the cancel log in here? (maybe a reply is ok? they are not a pair so it bother me.)

Reply:
lets add a server log for bgsave schedule? I feel like I'm leaning towards this.

Reply:
I do not mind placing a log in. case it is scheduled, but maybe it is important to distinguish the case it was explicitly requested by the user (ie BGAVE SCHEDULE) or when it was a result of BGSAVE call inside a transaction?

BTW - do you know why this support was introduced for transactions and not for scripts?

Reply:
bgsave is marked with noscript flag, so it is not allowed to execute in scripts.

Reply:
I know that, I wonder why it couldn't be done the same way as it is for multi/exec (means auto scheduled to be run after the script is executed)

Reply:
@enjoy-binbin done.